### PR TITLE
T-5 Create predefined data initializer.

### PR DESCRIPTION
**Pull request by goremukin**
Added database initializer, migrations deleted. Minor changes have been added to the data model. Fixed russian symbol 'С'.

### CODE REVIEW

**Comment by AntonBessarabov** [comment]
[T-5 Create predefined data initializer.](https://trello.com/c/EDCgEGA8/5-t-5-create-predefined-data-initializer)

File: PortfolioKeeper/Data/Context/FluentApiConfiguration/BankMap.cs [hunk 1]
```diff
@@ -12,6 +12,10 @@ public BankMap()
             Property(e => e.Name)
                 .HasColumnType("NVARCHAR")
                 .HasMaxLength(100);
+
+            Property(e => e.DispalayName)
```
Comment (AntonBessarabov) [inline PortfolioKeeper/Data/Context/FluentApiConfiguration/BankMap.cs]: Change to DisplayName, please

File: PortfolioKeeper/Data/Entities/Bank.cs [hunk 2]
```diff
@@ -3,5 +3,6 @@ namespace PortfolioKeeper.Data.Entities
     public class Bank : BaseEntity<byte>
     {
         public string Name { get; set; }
+        public string DispalayName { get; set; }
```
Comment (AntonBessarabov) [inline PortfolioKeeper/Data/Entities/Bank.cs]: Change to DisplayName, please

File: PortfolioKeeper/Data/Entities/BaseEntity.cs [hunk 3]
```diff
@@ -1,7 +1,11 @@
-﻿namespace PortfolioKeeper.Data.Entities
+﻿using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace PortfolioKeeper.Data.Entities
 {
     public class BaseEntity<TKey>
     {
+        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
```
Comment (AntonBessarabov) [inline PortfolioKeeper/Data/Entities/BaseEntity.cs]: Optional: We use only FluentApi, right?

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+            ParentAssetTypesInitializer(context);
+            RealtyAssetTypesInitializer(context);
+            AccountAssetTypesInitializer(context);
+            PreciousMetalsAssetTypesInitializer(context);
+        }
+
+        private void BanksInitialization(PortfolioKeeperDbContext context)
+        {
+            context.Banks.Add(new Bank { Name = "belgazprombank", DispalayName = "Белгазпромбанк" });
```
Comment (AntonBessarabov) [inline PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: Change to DisplayName, please

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+            context.Banks.Add(new Bank { Name = "bsbbank", DispalayName = "БСБ банк" });
```
Comment (AntonBessarabov) [inline PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: Change to DisplayName, please

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+            context.Banks.Add(new Bank { Name = "belarusbank", DispalayName = "Беларусбанк" });
```
Comment (AntonBessarabov) [inline PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: Change to DisplayName, please

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+            context.AssetTypeProperties.Add(new AssetPropertyType { AssetTypeId = 2, Name = AssertTypePropertiesConstants.AccountCountPropertyName, DisplayName = "Количество", CreatedOn = dateTime, UpdatedOn = dateTime });
+            context.AssetTypeProperties.Add(new AssetPropertyType { AssetTypeId = 3, Name = AssertTypePropertiesConstants.PreciousMetalsWeightPropertyName, DisplayName = "Вес", CreatedOn = dateTime, UpdatedOn = dateTime });
+            context.SaveChanges();
+        }
+
+        private void RealtyAssetTypesInitializer(PortfolioKeeperDbContext context)
+        {
+
+            context.AssetTypes.Add(new AssetType { ParrentTypeId = 1, Name = "apartment", DisplayName = "Квартира", CreatedOn = DateTime.Now });
+            context.AssetTypes.Add(new AssetType { ParrentTypeId = 1, Name = "car", DisplayName = "Автомобиль", CreatedOn = DateTime.Now });
```
Comment (AntonBessarabov) [inline PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: Mmm, do you mean broken cars?

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+            context.AssetTypeProperties.Add(new AssetPropertyType { AssetTypeId = 4, Name = AssertTypePropertiesConstants.ApartmentAreaPropertyName, DisplayName = "Площадь", DisplayOrder = 3, CreatedOn = dateTime, UpdatedOn = dateTime });
+
+            context.AssetTypeProperties.Add(new AssetPropertyType { AssetTypeId = 5, Name = AssertTypePropertiesConstants.CarBrandPropertyName, DisplayName = "Марка", DisplayOrder = 1, CreatedOn = dateTime, UpdatedOn = dateTime });
+            context.AssetTypeProperties.Add(new AssetPropertyType { AssetTypeId = 5, Name = AssertTypePropertiesConstants.CarModelPropertyName, DisplayName = "Модель", DisplayOrder = 2, CreatedOn = dateTime, UpdatedOn = dateTime });
+            context.AssetTypeProperties.Add(new AssetPropertyType { AssetTypeId = 5, Name = AssertTypePropertiesConstants.CarYearPropertyName, DisplayName = "Год выпуска", DisplayOrder = 3, CreatedOn = dateTime, UpdatedOn = dateTime });
```
Comment (AntonBessarabov) [inline PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: Fix this lines too, please

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+        {
+            context.Banks.Add(new Bank { Name = "belgazprombank", DispalayName = "Белгазпромбанк" });
+            context.Banks.Add(new Bank { Name = "bsbbank", DispalayName = "БСБ банк" });
+            context.Banks.Add(new Bank { Name = "belarusbank", DispalayName = "Беларусбанк" });
+            context.SaveChanges();
+        }
+
+        private void CurrenciesInitialization(PortfolioKeeperDbContext context)
+        {
+            context.Currencies.Add(new Currency { Name = "BYN", Description = "Новый белорусский рубль", CreatedOn = DateTime.Now });
```
Comment (AntonBessarabov) [inline PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: Change to Белорусский рубль, please

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+        }
+
+        private void ParentAssetTypesInitializer(PortfolioKeeperDbContext context)
+        {
+            context.AssetTypes.Add(new AssetType { Name = "realty", DisplayName = "Недвижимость", CreatedOn = DateTime.Now });
+            context.AssetTypes.Add(new AssetType { Name = "account", DisplayName = "Счёт", CreatedOn = DateTime.Now });
+            context.AssetTypes.Add(new AssetType { Name = "preciousMetals", DisplayName = "Драгоценные металлы", CreatedOn = DateTime.Now });
+            context.SaveChanges();
+
+            var dateTime = DateTime.UtcNow;
```
Comment (AntonBessarabov) [inline PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: Why you don't use UtcNow everywhere?

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+        }
+
+        private void RealtyAssetTypesInitializer(PortfolioKeeperDbContext context)
+        {
+
+            context.AssetTypes.Add(new AssetType { ParrentTypeId = 1, Name = "apartment", DisplayName = "Квартира", CreatedOn = DateTime.Now });
+            context.AssetTypes.Add(new AssetType { ParrentTypeId = 1, Name = "car", DisplayName = "Автомобиль", CreatedOn = DateTime.Now });
+            context.SaveChanges();
+
+            var dateTime = DateTime.UtcNow;
```
Comment (AntonBessarabov) [inline PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: Why you don't use UtcNow everywhere?

File: PortfolioKeeper/Data/Entities/BaseEntity.cs (hunk 3 above)
Comment (goremukin) [reply PortfolioKeeper/Data/Entities/BaseEntity.cs]: Aldeady no

File: PortfolioKeeper/Data/Context/FluentApiConfiguration/BankMap.cs (hunk 1 above)
Comment (goremukin) [reply PortfolioKeeper/Data/Context/FluentApiConfiguration/BankMap.cs]: fixed

File: PortfolioKeeper/Data/Entities/Bank.cs (hunk 2 above)
Comment (goremukin) [reply PortfolioKeeper/Data/Entities/Bank.cs]: fixed

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs (hunk 4 above)
Comment (goremukin) [reply PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: fixed

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs (hunk 9 above)
Comment (goremukin) [reply PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: fixed

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs (hunk 10 above)
Comment (goremukin) [reply PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: fixed

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs (hunk 11 above)
Comment (goremukin) [reply PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: fixed

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs (hunk 8 above)
Comment (goremukin) [reply PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: fixed

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs (hunk 5 above)
Comment (goremukin) [reply PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: fixed

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs (hunk 6 above)
Comment (goremukin) [reply PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: fixed

**Comment by AntonBessarabov** [comment]
Database tables are not generated correctly.
All string fields are nullable.
Add .IsRequired() for all of them in FluentApiConfiguration, please

**Comment by AntonBessarabov** [comment]
AssetMap.cs

- Add .IsRequired for Name property, please

AssetPropertyTypeMap.cs

- Remove duplicate of DisplayName property, please
- Add .IsRequired for Name and DisplayName properties, please

AssetPropertyValueMap.cs

- Add .IsRequired() for Value property, please

AssetTypeMap.cs

- Add .IsRequired for Name and DisplayName properties, please

BankMap.cs

- Add .IsRequired for Name and DisplayName  properties, please

CurrencyMap.cs

- Add .IsRequired for Name property, please
- Remove duplicate of Name property, please
- Add property Description:

```
            Property(e => e.Description)
                .IsRequired()
                .HasColumnType("NVARCHAR")
                .HasMaxLength(50);
```

MetalTypeMap.cs

- Add .IsRequired for Name and DisplayName  properties, please

PortfolioMap.cs

- Add .IsRequired for Name property, please

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs [hunk 12]
```diff
@@ -0,0 +1,175 @@
...
+                });
+
+            context.SaveChanges();
+        }
+
+        private void RealtyAssetTypesInitializer(PortfolioKeeperDbContext context)
+        {
+
+            context.AssetTypes.Add(new AssetType { ParrentTypeId = 1, Name = "apartment", DisplayName = "Квартира", CreatedOn = DateTime.UtcNow });
+            context.AssetTypes.Add(new AssetType { ParrentTypeId = 4, Name = "car", DisplayName = "Автомобиль", CreatedOn = DateTime.UtcNow });
```
Comment (AntonBessarabov) [inline PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: Why you init vehicle type in RealtyAssetTypesInitializer?

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs [hunk 13]
```diff
@@ -0,0 +1,175 @@
...
+                    UpdatedOn = dateTime });
+
+            context.AssetTypeProperties.Add(
+                new AssetPropertyType {
+                    AssetTypeId = 6,
+                    Name = AssertTypePropertiesConstants.CarBrandPropertyName,
+                    DisplayName = "Марка",
+                    DisplayOrder = 1,
+                    CreatedOn = dateTime,
+                    UpdatedOn = dateTime });
```
Comment (AntonBessarabov) [inline PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: Why you init vehicle type in RealtyAssetTypesInitializer?

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs [hunk 14]
```diff
@@ -0,0 +1,175 @@
...
+
+            context.AssetTypeProperties.Add(
+                new AssetPropertyType {
+                    AssetTypeId = 6,
+                    Name = AssertTypePropertiesConstants.CarModelPropertyName,
+                    DisplayName = "Модель",
+                    DisplayOrder = 2,
+                    CreatedOn = dateTime,
+                    UpdatedOn = dateTime });
```
Comment (AntonBessarabov) [inline PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: Why you init vehicle type in RealtyAssetTypesInitializer?

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs [hunk 15]
```diff
@@ -0,0 +1,175 @@
...
+
+            context.AssetTypeProperties.Add(
+                new AssetPropertyType {
+                    AssetTypeId = 6,
+                    Name = AssertTypePropertiesConstants.CarYearPropertyName,
+                    DisplayName = "Год выпуска",
+                    DisplayOrder = 3,
+                    CreatedOn = dateTime,
+                    UpdatedOn = dateTime });
```
Comment (AntonBessarabov) [inline PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: Why you init vehicle type in RealtyAssetTypesInitializer?

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs [hunk 16]
```diff
@@ -0,0 +1,175 @@
...
+        {
+            context.AssetTypes.Add(new AssetType { ParrentTypeId = 2, Name = "bynAccount", DisplayName = "Белорусские рубли", CreatedOn = DateTime.UtcNow });
+            context.AssetTypes.Add(new AssetType { ParrentTypeId = 2, Name = "usdAccount", DisplayName = "Доллары США", CreatedOn = DateTime.UtcNow });
+            context.AssetTypes.Add(new AssetType { ParrentTypeId = 2, Name = "eurAccount", DisplayName = "Евро", CreatedOn = DateTime.UtcNow });
+            context.SaveChanges();
+        }
+
+        private void PreciousMetalsAssetTypesInitializer(PortfolioKeeperDbContext context)
+        {
+            context.AssetTypes.Add(new AssetType { ParrentTypeId = 3, Name = "goldAssert", DisplayName = "Золото", CreatedOn = DateTime.UtcNow });
```
Comment (AntonBessarabov) [inline PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: I think that Name = "smthIngot" should be here

Comment (AntonBessarabov) [reply PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: And other display name too

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs (hunk 12 above)
Comment (goremukin) [reply PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: fixed

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs (hunk 13 above)
Comment (goremukin) [reply PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: fixed

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs (hunk 14 above)
Comment (goremukin) [reply PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: fixed

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs (hunk 15 above)
Comment (goremukin) [reply PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: fixed

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs (hunk 16 above)
Comment (goremukin) [reply PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: fixed

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs [hunk 17]
```diff
@@ -0,0 +1,175 @@
...
+
+namespace PortfolioKeeper.Data.Initializers
+{
+    public class PortfolioKeeperDbContextInitializer : CreateDatabaseIfNotExists<PortfolioKeeperDbContext>
+    {
+        protected override void Seed(PortfolioKeeperDbContext context)
+        {
+            BanksInitialization(context);
+            CurrenciesInitialization(context);
+
```
Comment (AntonBessarabov) [inline PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: Add here `MetalTypesInitialization(context);` , please

File: PortfolioKeeper/Data/Context/FluentApiConfiguration/PortfolioMap.cs [hunk 18]
```diff
@@ -11,7 +11,8 @@ public PortfolioMap()
 
             Property(e => e.Name)
                 .HasColumnType("NVARCHAR")
-                .HasMaxLength(30);
+                .HasMaxLength(30)
```
Comment (AntonBessarabov) [inline PortfolioKeeper/Data/Context/FluentApiConfiguration/PortfolioMap.cs]: Change to `.HasMaxLength(50)`, please

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs (hunk 17 above)
Comment (goremukin) [reply PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: fixed

File: PortfolioKeeper/Data/Context/FluentApiConfiguration/PortfolioMap.cs (hunk 18 above)
Comment (goremukin) [reply PortfolioKeeper/Data/Context/FluentApiConfiguration/PortfolioMap.cs]: fixed

File: PortfolioKeeper/Data/Entities/AssetType.cs [hunk 19]
```diff
@@ -7,6 +7,7 @@ public class AssetType : BaseEntity<byte>
         public byte? ParrentTypeId { get; set; }
 
         public string Name { get; set; }
+        public string DisplayName { get; set; }
```
Comment (stas-litvinenka) [inline PortfolioKeeper/Data/Entities/AssetType.cs]: add empty row here

File: PortfolioKeeper/Data/Entities/Bank.cs [hunk 20]
```diff
@@ -3,5 +3,6 @@ namespace PortfolioKeeper.Data.Entities
     public class Bank : BaseEntity<byte>
     {
         public string Name { get; set; }
+        public string DisplayName { get; set; }
```
Comment (stas-litvinenka) [inline PortfolioKeeper/Data/Entities/Bank.cs]: empty row here

File: PortfolioKeeper/Data/Entities/BaseEntity.cs (hunk 3 above)
Comment (stas-litvinenka) [reply PortfolioKeeper/Data/Entities/BaseEntity.cs]: please redo like this - .Property(p => p.Id).HasDatabaseGeneratedOption(DatabaseGeneratedOption.Identity); - remove data annotation here and add in fluent api

File: PortfolioKeeper/Data/Initializers/AssertTypePropertiesConstants.cs [hunk 21]
```diff
@@ -0,0 +1,18 @@
+﻿namespace PortfolioKeeper.Data.Initializers
+{
+    public class AssertTypePropertiesConstants
+    {
+        public const string RealtyPricePropertyName = "realtyPrice";
```
Comment (stas-litvinenka) [inline PortfolioKeeper/Data/Initializers/AssertTypePropertiesConstants.cs]: use UpperCase here e.g. - RealtyPrice, Count , etc.

Comment (stas-litvinenka) [inline PortfolioKeeper/Data/Initializers/AssertTypePropertiesConstants.cs]: Better split to different classes and make rename. e.g. - RealtyProperties(class name). Price(PropertyName)

File: PortfolioKeeper/Data/Initializers/AssertTypePropertiesConstants.cs [hunk 22]
```diff
@@ -0,0 +1,18 @@
...
+        public const string AccountCountPropertyName = "count";
```
Comment (stas-litvinenka) [inline PortfolioKeeper/Data/Initializers/AssertTypePropertiesConstants.cs]: Same here and below

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs [hunk 23]
```diff
@@ -0,0 +1,190 @@
...
+            RealtyAssetTypesInitializer(context);
+            VehicleAssetTypesInitializer(context);
+            AccountAssetTypesInitializer(context);
+            PreciousMetalsAssetTypesInitializer(context);
+            MetalTypesInitialization(context);
+        }
+
+        private void BanksInitialization(PortfolioKeeperDbContext context)
+        {
+            context.Banks.Add(new Bank { Name = "belgazprombank", DisplayName = "Белгазпромбанк" });
```
Comment (stas-litvinenka) [inline PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: Use Uppercase .e.g. Belgazprombank

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs [hunk 24]
```diff
@@ -0,0 +1,190 @@
...
+        {
+            context.Currencies.Add(new Currency { Name = "BYN", Description = "Белорусский рубль", CreatedOn = DateTime.UtcNow });
+            context.Currencies.Add(new Currency { Name = "USD", Description = "Доллар США", CreatedOn = DateTime.UtcNow });
+            context.Currencies.Add(new Currency { Name = "EUR", Description = "Евро", CreatedOn = DateTime.UtcNow });
+            context.SaveChanges();
+        }
+
+        private void MetalTypesInitialization(PortfolioKeeperDbContext context)
+        {
+            context.MetalTypes.Add(new MetalType { Name = "gold", DisplayName = "Золото" });
```
Comment (stas-litvinenka) [inline PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: Uppercase

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs [hunk 25]
```diff
@@ -0,0 +1,190 @@
...
+            context.MetalTypes.Add(new MetalType { Name = "silver", DisplayName = "Серебро" });
+            context.MetalTypes.Add(new MetalType { Name = "platinum", DisplayName = "Платина" });
+
```
Comment (stas-litvinenka) [inline PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: be consistent - either remove empty line or add everywhere before context.SaveChanges()

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs [hunk 26]
```diff
@@ -0,0 +1,190 @@
...
+            context.SaveChanges();
+        }
+
+        private void ParentAssetTypesInitializer(PortfolioKeeperDbContext context)
+        {
+            context.AssetTypes.Add(new AssetType { Name = "realty", DisplayName = "Недвижимость", CreatedOn = DateTime.UtcNow });
```
Comment (stas-litvinenka) [inline PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: UpperCase

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs [hunk 27]
```diff
@@ -0,0 +1,190 @@
...
+            context.AssetTypes.Add(new AssetType { Name = "account", DisplayName = "Счёт", CreatedOn = DateTime.UtcNow });
+            context.AssetTypes.Add(new AssetType { Name = "preciousMetals", DisplayName = "Драгоценные металлы", CreatedOn = DateTime.UtcNow });
+            context.AssetTypes.Add(new AssetType { Name = "vehicle", DisplayName = "Транспортные средства", CreatedOn = DateTime.UtcNow });
+            context.SaveChanges();
+
+            var dateTime = DateTime.UtcNow;
```
Comment (stas-litvinenka) [inline PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: put it above  - so all data will have the same time- it's better than call each time Datetime.Utcnow

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs [hunk 28]
```diff
@@ -0,0 +1,190 @@
...
+
+            context.AssetTypeProperties.Add(
+                new AssetPropertyType {
```
Comment (stas-litvinenka) [inline PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: Use better formatting. looks strange now. js style

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs [hunk 29]
```diff
@@ -0,0 +1,190 @@
...
+                    CreatedOn = dateTime,
+                    UpdatedOn = dateTime
+                });
+
+            context.SaveChanges();
+        }
+
+        private void RealtyAssetTypesInitializer(PortfolioKeeperDbContext context)
+        {
+
```
Comment (stas-litvinenka) [inline PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: remove empty line

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs [hunk 30]
```diff
@@ -0,0 +1,190 @@
...
+            context.AssetTypes.Add(new AssetType { ParrentTypeId = 1, Name = "apartment", DisplayName = "Квартира", CreatedOn = DateTime.UtcNow });
```
Comment (stas-litvinenka) [inline PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: UpperCase.

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs [hunk 31]
```diff
@@ -0,0 +1,190 @@
...
+            context.SaveChanges();
+
+            var dateTime = DateTime.UtcNow;
```
Comment (stas-litvinenka) [inline PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: move above

File: PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs [hunk 32]
```diff
@@ -0,0 +1,190 @@
...
+                    CreatedOn = dateTime,
+                    UpdatedOn = dateTime
+                });
+
+            context.SaveChanges();
+        }
+
+        private void AccountAssetTypesInitializer(PortfolioKeeperDbContext context)
+        {
+            context.AssetTypes.Add(new AssetType { ParrentTypeId = 2, Name = "bynAccount", DisplayName = "Белорусские рубли", CreatedOn = DateTime.UtcNow });
```
Comment (stas-litvinenka) [inline PortfolioKeeper/Data/Initializers/PortfolioKeeperDbContextInitializer.cs]: uppercase

File: PortfolioKeeper/Data/Initializers/AssertTypePropertiesConstants.cs [hunk 33]
```diff
@@ -0,0 +1,18 @@
+﻿namespace PortfolioKeeper.Data.Initializers
+{
+    public class AssertTypePropertiesConstants
```
Comment (stas-litvinenka) [inline PortfolioKeeper/Data/Initializers/AssertTypePropertiesConstants.cs]: Please create common class Constants, create AssetTypeProperties class inside it

**Pull request merged**